src/fairchem/core/datasets/base_dataset.py

```diff
@@ -0,0 +1,227 @@
+
+    Args:
+        config (dict): dataset config dictionary
+        split (str): name of split
+
+    Returns:
+        Subset: dataset subset class
+    """
+    # Initialize the dataset
+    dataset_cls = registry.get_dataset_class(config.get("format", "lmdb"))
+    assert issubclass(dataset_cls, Dataset), f"{dataset_cls} is not a Dataset"
+
+    # remove information about other splits, only keep specified split
+    # this may only work with the mt config not main config
+    current_split_config = config.copy()
+    if "splits" in current_split_config:
+        current_split_config.pop("splits")
+        current_split_config.update(config["splits"][split])
+
+    seed = current_split_config.get("seed", 0)
+    if split != "train":
+        seed += (
+            1  # if we use same dataset for train / val , make sure its diff sampling
+        )
+
+    g = torch.Generator()
+    g.manual_seed(seed)
+
+    dataset = dataset_cls(current_split_config)
+    # Get indices of the dataset
+    indices = dataset.indices
+    max_atoms = current_split_config.get("max_atoms", None)
+    if max_atoms is not None:
+        if not dataset.metadata_hasattr("natoms"):
+            raise ValueError("Cannot use max_atoms without dataset metadata")
+        indices = indices[dataset.get_metadata("natoms", indices) <= max_atoms]
+
+    # Apply dataset level transforms
+    # TODO is no_shuffle mutually exclusive though? or what is the purpose of no_shuffle?
+    first_n = current_split_config.get("first_n")
+    sample_n = current_split_config.get("sample_n")
+    no_shuffle = current_split_config.get("no_shuffle")
+    # this is true if at most one of the mutually exclusive arguments are set
+    if sum(arg is not None for arg in (first_n, sample_n, no_shuffle)) > 1:
+        raise ValueError(
+            "sample_n, first_n, no_shuffle are mutually exclusive arguments. Only one can be provided."
+        )
+    if first_n is not None:
+        max_index = first_n
+    elif sample_n is not None:
+        # shuffle by default, user can disable to optimize if they have confidence in dataset
+        # shuffle all datasets by default to avoid biasing the sampling in concat dataset
+        # TODO only shuffle if split is train
+        max_index = sample_n
+        indices = indices[randperm(len(indices), generator=g)]
+    else:
+        max_index = len(indices)
+        indices = (
+            indices if no_shuffle else indices[randperm(len(indices), generator=g)]
+        )
+
+    if max_index > len(indices):
+        msg = (
+            f"Cannot take {max_index} data points from a dataset of only length {len(indices)}.\n"
+            f"Make sure to set first_n or sample_n to a number =< the total samples in dataset."
+        )
+        if max_atoms is not None:
+            msg = msg[:-1] + f"that are smaller than the given max_atoms {max_atoms}."
+        raise ValueError(msg)
+
+    indices = indices[:max_index]
```

Comment:
I'm testing this on my end...If i have a config as follows:

```
dataset:
  train:
    format: lmdb
    src: data/s2ef/all/train/
    first_n: 2000000
    key_mapping:
      y: energy
      force: forces
    transforms:
      normalizer:
        energy:
          mean: -0.7554450631141663
          stdev: 2.887317180633545
        forces:
          mean: 0
          stdev: 2.887317180633545
  val:
    src: data/s2ef/all/val_id_30k
```

Because we have val set up to inherit the train parameters, it will also try to grab `first_n` for validation/testing as well. To get around this I would need to explicitly define `first_n` for the validation/test splits which is not ideal. 

We should have it such that something like `first_n` does not try to get passed down to val/test. I don't know the best way we want to tackle this, open to ideas.

Reply:
Same is true with `metadata_path`, etc. All of it is trying to be inherited

Reply:
Totally fair, again great catch @mshuaibii ! Thank you! Looking into it...